saleor/graphql/webhook/mutations.py

```diff
@@ -301,3 +311,95 @@ def perform_mutation(cls, _root, info, **data):
+    )
+
+    class Arguments:
+        id = graphene.ID(description="The ID of the webhook.", required=True)
+        input = WebhookDryRunInput(
+            description="Fields required to perform dry run of a webhook event.",
+            required=True,
+        )
+
+    class Meta:
+        description = "Performs a dry run of a webhook event."
+        permissions = (AppPermission.MANAGE_APPS,)
+        error_type_class = WebhookDryRunError
+
+    @classmethod
+    def validate_input(cls, info, **data):
+        query = data.get("input", {}).get("query")
+        object_id = data.get("input", {}).get("object_id")
+        webhook_id = data.get("id")
+
+        event_type = get_event_type_from_subscription(query)
+        if not event_type:
+            raise_validation_error(
+                field="query",
+                message="Can't parse an event type from query.",
+                code=WebhookDryRunErrorCode.GRAPHQL_ERROR,
+            )
+
+        event = WEBHOOK_TYPES_MAP.get(event_type)
+        if not event:
+            event_name = event_type[0].upper() + to_camel_case(event_type)[1:]
+            raise_validation_error(
+                field="query",
+                message=f"Event type: {event_name} is not defined in graphql schema.",
+                code=WebhookDryRunErrorCode.GRAPHQL_ERROR,
+            )
+
+        related_models = [
+            field.type.get_model().__name__
+            for field in event._meta.fields.values()
+            if hasattr(field, "type") and hasattr(field.type, "get_model")
+        ]
+        model, _ = graphene.Node.from_global_id(object_id)
+        if model not in related_models:
+            raise_validation_error(
+                field="objectId",
+                message="ObjectId doesn't match event type.",
+                code=WebhookDryRunErrorCode.INVALID_ID,
+            )
+
+        object = cls.get_node_or_error(info, object_id, field="objectId")
+        webhook = cls.get_node_or_error(info, webhook_id, field="id")
+        app = webhook.app
+
+        if permission := WebhookEventAsyncType.PERMISSIONS.get(event_type):
+            codename = permission.value.split(".")[1]
+            app_permissions = [perm.codename for perm in app.permissions.all()]
+            if codename not in app_permissions:
+                raise_validation_error(
+                    message=f"The app doesn't have required permission: {codename}.",
+                    code=WebhookDryRunErrorCode.MISSING_APP_PERMISSION,
+                )
+
+        return event_type, object, app, query
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        event_type, object, app, query = cls.validate_input(info, **data)
+        payload = None
+        if all([event_type, object, app, query]):
+            request = initialize_request()
+            payload = generate_payload_from_subscription(
+                event_type, object, query, request, app
```

Comment:
As we discussed on Slack with @patrys we want to run dry-run as the staff user and not as the app. Then we would have to remove the `app` from this call I think.